src/Nethermind/Nethermind.Synchronization.Test/SyncServerTests.cs

```diff
@@ -319,16 +314,41 @@ public void Can_inject_terminal_block_with_higher_td_than_head(long ttd, bool se
             }
 
             ctx.SyncServer.AddNewBlock(block, ctx.NodeWhoSentTheBlock);
-            Assert.AreEqual(localBlockTree.BestSuggestedHeader!.Number, 9);
-            localBlockTree.FindBlock(terminalBlockWithHigherTotalDifficulty.Hash!, BlockTreeLookupOptions.None).Should().NotBeNull();
-            localBlockTree.BestSuggestedHeader!.Hash.Should().Be(terminalBlockWithHigherTotalDifficulty.Hash!);
+            Assert.AreEqual(ctx.LocalBlockTree.BestSuggestedHeader!.Number, 9);
+            ctx.LocalBlockTree.FindBlock(terminalBlockWithHigherTotalDifficulty.Hash!, BlockTreeLookupOptions.None).Should().NotBeNull();
+            ctx.LocalBlockTree.BestSuggestedHeader!.Hash.Should().Be(terminalBlockWithHigherTotalDifficulty.Hash!);
         }
 
-        private Context CreateMergeContext(IBlockTree localBlockTree, UInt256 ttd)
+
+        [TestCase(9000010)]
+        public void PostMerge_block_from_gossip_should_not_override_main_chain(long ttd)
+        {
+            BlockTree remoteBlockTree = Build.A.BlockTree().OfChainLength(9).TestObject;
+            Block poWBlockPostMerge = Build.A.Block.WithDifficulty(1000010).WithParent(remoteBlockTree.Head).WithTotalDifficulty(remoteBlockTree.Head.TotalDifficulty + 1000010).TestObject;
+            remoteBlockTree.SuggestBlock(poWBlockPostMerge);
+
+            Context ctx = CreateMergeContext(10, (UInt256)ttd);
+            Block newPostMergeBlock = Build.A.Block.WithDifficulty(0).WithParent(ctx.LocalBlockTree.Head).WithTotalDifficulty(ctx.LocalBlockTree.Head.TotalDifficulty).TestObject;
+            ctx.LocalBlockTree.SuggestBlock(newPostMergeBlock);
+            ctx.LocalBlockTree.UpdateMainChain(new[] {newPostMergeBlock} , true, true );
+
+            Block block = remoteBlockTree.FindBlock(9, BlockTreeLookupOptions.None);
```

Comment:
NP: Why does it need to get back the block from remoteBlockTree? Why not pass poWBlockPostMerge?

Reply:
hmmm... I don't get it. Let's discuss it in any occasion :)